src/arr/compiler/well-formed.arr

```diff
@@ -992,228 +999,25 @@ top-level-visitor = A.default-iter-visitor.{
-  end,
-  method s-tuple-get(_, l :: Loc, tup, index, index-loc):
-    well-formed-visitor.s-tuple-get(l, tup, index, index-loc)
-  end,
-  method s-obj(_, l :: Loc, fields :: List<A.Member>):
-    well-formed-visitor.s-obj(l, fields)
-  end,
-  method s-array(_, l :: Loc, values :: List<A.Expr>):
-    well-formed-visitor.s-array(l, values)
-  end,
-  method s-construct(_, l :: Loc, mod :: A.ConstructModifier, constructor :: A.Expr, values :: List<A.Expr>):
-    well-formed-visitor.s-construct(l, mod, constructor, values)
-  end,
-  method s-app(_, l :: Loc, _fun :: A.Expr, args :: List<A.Expr>):
-    well-formed-visitor.s-app(l, _fun, args)
-  end,
-  method s-prim-app(_, l :: Loc, _fun :: String, args :: List<A.Expr>):
-    well-formed-visitor.s-prim-app(l, _fun, args)
-  end,
-  method s-frac(_, l :: Loc, num, den):
-    well-formed-visitor.s-frac(l, num, den)
-  end,
-  method s-rfrac(_, l :: Loc, num, den):
-    well-formed-visitor.s-rfrac(l, num, den)
-  end,
-  method s-id(_, l :: Loc, id :: A.Name):
-    well-formed-visitor.s-id(l, id)
-  end,
-  method s-id-var(_, l :: Loc, id :: A.Name):
-    well-formed-visitor.s-id-var(l, id)
-  end,
-  method s-id-letrec(_, l :: Loc, id :: A.Name, safe :: Boolean):
-    well-formed-visitor.s-id-letrec(l, id, safe)
-  end,
-  method s-dot(_, l :: Loc, obj :: A.Expr, field :: String):
-    well-formed-visitor.s-dot(l, obj, field)
-  end,
-  method s-get-bang(_, l :: Loc, obj :: A.Expr, field :: String):
-    well-formed-visitor.s-get-bang(l, obj, field)
-  end,
-  method s-bracket(_, l :: Loc, obj :: A.Expr, field :: A.Expr):
-    well-formed-visitor.s-bracket(l, obj, field)
-  end,
-  method s-for(_, l :: Loc, iterator :: A.Expr, bindings :: List<A.ForBind>, ann :: A.Ann, body :: A.Expr, blocky :: Boolean):
-    well-formed-visitor.s-for(l, iterator, bindings, ann, body, blocky)
-  end,
-  method s-check(_, l :: Loc, name :: Option<String>, body :: A.Expr, keyword-check :: Boolean):
-    well-formed-visitor.s-check(l, name, body, keyword-check)
-  end,
-  method s-data-field(_, l :: Loc, name :: String, value :: A.Expr):
-    well-formed-visitor.s-data-field(l, name, value)
-  end,
-  method s-mutable-field(_, l :: Loc, name :: String, ann :: A.Ann, value :: A.Expr):
-    well-formed-visitor.s-mutable-field(l, name, ann, value)
-  end,
-  method s-method-field(_, l :: Loc, name :: String, params :: List<A.Name>, args :: List<A.Bind>, ann :: A.Ann, doc :: String, body :: A.Expr, _check-loc :: Option<Loc>, _check :: Option<A.Expr>, blocky):
-    well-formed-visitor.s-method-field(l, name, params, args, ann, doc, body, _check-loc, _check, blocky)
+    # TODO
+    true
   end,
-  method s-for-bind(_, l :: Loc, bind :: A.Bind, value :: A.Expr):
-    well-formed-visitor.s-for-bind(l, bind, value)
+  method s-app(self, l :: Loc, _fun :: A.Expr, args :: List<A.Expr>) block:
+    if (A.is-s-dot(_fun) and A.is-s-id(_fun.obj)
+        and (_fun.obj.id.toname() == "builtins") and (_fun.field == "trace-value")):
```

Comment:
Yeah :(  This one's necessary because we allow `type-let` as a block-ending expression, but only at the top level.  But `trace-value` will wrap all top-level, block-ending expressions...which makes the type-let no longer be a top-level expression!  (At least we had a regression test for this...)